### PR TITLE
Make lavaland bridges hover over rivers.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR moves Lavaland bridge tiles above their rivers.

The bridge tiles are destructible.

## Why It's Good For The Game
Looks cool, has some interesting implications allowing people to take out bridges.
## Images of changes

2024_11_24__17_01_08__Paradise Station 13
2024_11_24__17_01_15__Paradise Station 13

## Testing

See above

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
tweak: Lavaland bridges are now raised above the lava.
/:cl:

### CODE REVIEW

File: code/datums/ruins/bridges/bridges.dm [hunk 1]
```diff
@@ -109,6 +113,62 @@
...
+	. = ..()
+	add_shadow = add_shadow_
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/bridge_walkway/LateInitialize()
+	. = ..()
+	var/turf/T = get_turf(src)
+	if(T && add_shadow)
+		T.layer = LATTICE_LAYER // prevent occlusion of drop shadows from bordering turfs
+		T.add_filter("AO", 1, drop_shadow_filter(x = 0, y = 0, size = 5, offset = 0, color = "#04080FEE"))
```
Comment (Contrabang) [inline code/datums/ruins/bridges/bridges.dm:134]: Not sure the best way to handle this, but doesn't respect the `mymob.client.prefs.toggles & PREFTOGGLE_AMBIENT_OCCLUSION` pref

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/datums/ruins/bridges/bridges.dm [hunk 2]
```diff
@@ -109,6 +113,62 @@
...
+
+/obj/structure/bridge_walkway/Destroy()
+	var/turf/T = get_turf(src)
+	if(T)
+		T.layer = PLATING_LAYER
+		T.clear_filters()
+
+	. = ..()
```
Comment (lewcc) [inline code/datums/ruins/bridges/bridges.dm:142]:
```suggestion
	return ..()
```

Comment (warriorstar-orion) [reply code/datums/ruins/bridges/bridges.dm]: Fixed

File: code/datums/ruins/bridges/bridges.dm (hunk 1 above)
Comment (warriorstar-orion) [reply code/datums/ruins/bridges/bridges.dm]: I didn't realize how prominent the default AO for everything is, I'll just let that affect the appearance instead.

**Review by Burzah — APPROVED** [review]
Looks good!

**Review by Contrabang — APPROVED** [review]

**Pull request merged**